File: 이재창/Java/singleTone/src/singleTone/SingleTest.java
package singleTone;

public class SingleTest {
	private static SingleTest inst;
	private int data;
	
	public SingleTest() {
		data = (int)(Math.random()*100);
	}
	
	public static SingleTest getInstance() {
		if (inst == null) {
			inst = new SingleTest();
		}
		return inst;
	}
	
	public int getData() {
		return data;
	}
}

File: 강병수/C#/210215/Problem3/Problem3/SingleTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Problem3
{
    class SingleTest
    {
        private static SingleTest inst;
        private int data;
        Random ran = new Random();
        
        public SingleTest()
        {
            data = (int)(ran.NextDouble() * 100);
        }

        public static SingleTest getInstance()
        {
            if(inst == null)
            {
                inst = new SingleTest();
            }
            return inst;
        }

        public int getData()
        {
            return data;
        }
    }
}
